dask/dataframe/groupby.py

```diff
@@ -1056,38 +1052,38 @@ def __init__(
         self.group_keys = group_keys
         self.obj = df
         # grouping key passed via groupby method
-        self.index = _normalize_index(df, by)
+        self.by = _normalize_by(df, by)
         self.sort = sort
 
-        if isinstance(self.index, list):
-            do_index_partition_align = all(
+        if isinstance(self.by, list):
+            do_by_partition_align = all(
                 item.npartitions == df.npartitions if isinstance(item, Series) else True
-                for item in self.index
+                for item in self.by
             )
-        elif isinstance(self.index, Series):
-            do_index_partition_align = df.npartitions == self.index.npartitions
+        elif isinstance(self.by, Series):
+            do_by_partition_align = df.npartitions == self.by.npartitions
         else:
-            do_index_partition_align = True
+            do_by_partition_align = True
 
-        if not do_index_partition_align:
+        if not do_by_partition_align:
             raise NotImplementedError(
-                "The grouped object and index of the "
+                "The grouped object and by of the "
```

Comment:
```suggestion
                "The grouped object and `by` of the "
```
Quoting this in errors/docstrings will especially help readability with the new name